hoot-core/src/main/cpp/hoot/core/conflate/DuplicateWayRemover.cpp

```diff
@@ -131,13 +135,25 @@ void DuplicateWayRemover::apply(shared_ptr<OsmMap>& map)
           LOG_VARD(w->getTags());
           LOG_VARD(w2->getTags());
 
-          if (w->getNodeCount() > w2->getNodeCount())
-          {
-            _removeDuplicateNodes(w, _map->getWay(wit->first));
-          }
-          else
+          double textTagScore = 0.0;
+          double enumTagScore = 0.0;
+          double weight = 0.0;
+          //If all non-name tags aren't exactly the same between these two ways (or the strict
+          //tag matching isn't activated), then we won't merge their geometries or tags.
+          TagComparator::getInstance().compareTextTags(w->getTags(), w2->getTags(), textTagScore, weight);
+          TagComparator::getInstance().compareEnumeratedTags(w->getTags(), w2->getTags(), enumTagScore, weight);
```

Comment:
If you're trying to do a strict tag comparison then I'd check all the tags for an exact match that are not metadata. (OsmSchema::isMetaData). The TagComparator functions are intentionally fuzzy and I don't think you want fuzzy in this code with strict matching. You should also probably only calculate the tag matches if strictTagMatching is true -- those tag comparisons can be a little expensive sometimes.